cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDTransactions.java

```diff
@@ -0,0 +1,139 @@
+package com.skillstorm.StepDefinitions;
+
+
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+
+import com.skillstorm.WebDriverSingleton;
+import com.skillstorm.PageObjects.LoginPage;
+import com.skillstorm.PageObjects.TransactionPage;
+import com.skillstorm.Utilities.Navigator;
+import com.skillstorm.Utilities.UserData.User;
+import com.skillstorm.Utilities.UserData.UserType;
+
+import java.time.Duration;
+
+import org.openqa.selenium.WebDriver;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.*;
+
+public class SDTransactions {
+
+    WebDriver driver;
+    WebDriverWait wait;
+    Navigator navigator;
+    private TransactionPage transactions;
+    private String[] transactionContent;
+
+    /**
+     * Set up the ChromeDriver and initialize TransactionPage.
+     */
+    @Before("@transaction")
+    public void setUp() {
+        driver = WebDriverSingleton.getDriver();
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        navigator = new Navigator(driver);
+
+        this.transactions = new TransactionPage(driver);
+
+        transactionContent =  new String[4];
+    }
+
+    /**
+     * Tear down the WebDriver after each test.
+     */
+    @After("@transaction")
+    public void tearDown() {
+        WebDriverSingleton.quitDriver();
+    }
+
+    /**
+     * Login Scenario Definitions
+     */
+    @Given("I login")
+    public void login() {
+        navigator.navigateTo(Navigator.PGNAME_LOGIN);
+        LoginPage loginPage = new LoginPage(driver);
+        User user = new User(UserType.PERSISTANT, "joseph.sam@gmail.com", "password1");
+        loginPage.login(user);
+    }
+
+    /**
+     * General Scenario Definitions
+     */
+    @Given("I am on the Transactions page")
+    public void iAmOnTheTranactionsPage() {
+        this.transactions.clickTab();
+        String expectedUrl = "http://localhost:5173/dashboard/transactions";
+        String actualUrl = this.transactions.getCurrentUrl();
+        Assert.assertEquals(actualUrl, expectedUrl);
+    }
+
+    /**
+     * Create Transaction Scenario Definitions
+     */
+
+    @And("I fill in the {string}, {string}, {string}, and {string}")
+    public void andIFillInThe(String name, String account, String amount, String category) {
+        transactionContent[0] = this.transactions.setName(name);
+        transactionContent[1] = this.transactions.setAccount(account);
+        transactionContent[2] = this.transactions.setAmount(amount);
+        transactionContent[3] = this.transactions.setCategory(category);
+    }
+
+    @Then("I can see the new transaction in my list")
+    public void iCanSeeTheNewTransactionInMyList() {
```

Comment:
Low impact: This Then step does not ensure that anything happens.
Suggestion: Remove or add an assertion of some sort.